File: py/560.py
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        sum=ret=0;
        m_map=collections.Counter();
        m_map[0]=1;
        for num in nums:
            sum+=num;
            ret+=m_map[sum-k];
            m_map[sum]+=1;
        return ret;

File: C++/560.cpp
#include"mod.h"
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int, int>m_map;
        m_map[0]=1;
        int sum=0, ret=0;
        for(int i=0; i<nums.size(); ++i){
            sum+=nums[i];
            ret+=m_map[sum-k];
            ++m_map[sum];
        }
        return ret;
    }
};